PROTOCOL.md

```diff
@@ -456,17 +467,29 @@ Protocol versioning allows a newer client to exclude older readers and/or writer
 The _protocol version_ will be increased whenever non-forward-compatible changes are made to this specification.
 In the case where a client is running an invalid protocol version, an error should be thrown instructing the user to upgrade to a newer protocol version of their Delta client library.
 
-Since breaking changes must be accompanied by an increase in the protocol version recorded in a table, clients can assume that unrecognized fields or actions are never required in order to correctly interpret the transaction log.
+Since breaking changes must be accompanied by an increase in the protocol version recorded in a table or by the addition of a Table Feature, clients can assume that unrecognized fields or actions are never required in order to correctly interpret the transaction log. Clients must ignore such unrecognized fields, and should not produce an error when reading a table that contains unrecognized fields.
+
+Reader Version 3 and Writer Version 7 add two lists of Table Features to the protocol action. The capability for readers and writers to operate on such a table is not only dependent on their supported protocol versions, but also on whether they support all features listed in `readerFeatures` and `writerFeatures`. See [Table Features](#table-features) section for more information.
 
 The schema of the `protocol` action is as follows:
 
 Field Name | Data Type | Description
 -|-|-
 minReaderVersion | Int | The minimum version of the Delta read protocol that a client must implement in order to correctly *read* this table
 minWriterVersion | Int | The minimum version of the Delta write protocol that a client must implement in order to correctly *write* this table
+readerFeatures | Array[Feature Struct] | A collection of features that a client must implement in order to correctly read this table (exist only when `minReaderVersion` is set to `3`)
+writerFeatures | Array[Feature Struct] | A collection of features that a client must implement in order to correctly write this table (exist only when `minWriterVersion` is set to `7`)
 
-The current version of the Delta protocol is:
-```
+#### Feature Struct
+
+Field Name | Data Type | Description
+-|-|-
+name | String | A globally unique _feature name_
+status | String | Indicates whether the table currently uses the feature. The only valid value is `enabled`.
```

Comment:
what's the point of having a **required** field with only 1 allowed value?
e.g. if just `name` is present, then that could imply that the feature is enabled.

Reply:
Thanks for the feedback! It does make sense to remove this `status` as it's too hard to add another status with proper backward compatibility (not breaking checkpoints etc.). We just merged https://github.com/delta-io/delta/commit/5c2e6d040092969898f1881c7d95ec278d0b108d to address this suggestion.